### ISSUE TITLE
Marshal exception for oci_objectstorage_preauthrequest.tomcat upgrading provider version 3.19 to 3.20

### ISSUE THREAD

**Issue by tomzepf-oracle** [terraform-providers/terraform-provider-oci#748]
### Terraform Version

Terraform v0.11.13
+ provider.oci v3.20.0

### OCI Provider Version

+ provider.oci v3.20.0

### Description:

1. Create an OS object named "tomcat/sample.war" in the "GBUCA_Shared_Download" bucket - contents not important.
1. Edit the local variables at the beginning of the provided terraform file for tenancy, 
1. Set provider version to "<= 3.19" in the provided terraform file.
1. terraform init, terraform apply.
1. Set provider version to "<= 3.20" (or clear the version attribute)
1. terraform init -upgrade, terraform apply

You will get this error output:

```
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.oci_objectstorage_bucket.download: Refreshing state...
data.oci_objectstorage_objects.tomcat-sample-war: Refreshing state...
oci_objectstorage_preauthrequest.tomcat-sample-war: Refreshing state... (ID: 2SzSQBgUCXPblJHfFMjtDV+r3Xbp40u8X/SVy6CP1XM=:tomcat/sample.war)

Error: Error refreshing state: 1 error(s) occurred:

* oci_objectstorage_preauthrequest.tomcat-sample-war: 1 error(s) occurred:

* oci_objectstorage_preauthrequest.tomcat-sample-war: oci_objectstorage_preauthrequest.tomcat-sample-war: can not marshal to path in request for field ParId. Due to can not marshal a nil pointer
```

Note, this does not reproduce if you start with provider version >= 3.20.

[par-test.zip](https://github.com/terraform-providers/terraform-provider-oci/files/3049142/par-test.zip)

**Comment by parrneet** [comment]
Hey @tomzepf-oracle - We have the fix for this in place and expect this to be working in next release.

**Comment by alexng-canuck** [comment]
Hi @tomzepf-oracle , this issue should now be fixed as of v3.22.0. See https://github.com/terraform-providers/terraform-provider-oci/blob/master/CHANGELOG.md#3220-april-10-2019 for more details.
